crates/nu-parser/tests/test_parser.rs

```diff
@@ -1043,6 +1043,34 @@ fn test_semi_open_brace(#[case] phrase: &[u8]) {
     let _block = parse(&mut working_set, None, phrase, true);
 }
 
+#[rstest]
+#[case(b"def test []: int -> int { 1 }")]
+#[case(b"def test []: string -> string { 'qwe' }")]
+#[case(b"def test []: nothing -> nothing { null }")]
+#[case(b"def test []: list<string> -> list<string> { [] }")]
+#[case(b"def test []: record<a: int b: int> -> record<c: int e: int> { {c: 1 e: 1} }")]
+#[case(b"def test []: table<a: int b: int> -> table<c: int e: int> { [ {c: 1 e: 1} ] }")]
+fn test_input_signature(#[case] phrase: &[u8]) {
```

Comment:
could you maybe add a test case where the `table` or `record` annotation is left unterminated?

```nushell
def foo []: record<bar: int {}
```

Reply:
Done.
Tests for type annotations can now be positive(testing for success) and negative(testing for failure)
Also discovered that individual parametrized  cases in `rstest`  can be named, so I merged input/output and let tests together